state-chain/custom-rpc/src/lib.rs

```diff
@@ -1421,6 +1426,7 @@ where
 		to_asset: Asset,
 		amount: U256,
 		broker_commission: BasisPoints,
+		boost_fee: BasisPoints,
```

Comment:
this will allow the caller to pass the expected boost fee in basis points, right?

Reply:
for us it would be awesome to just pass a boolean if the quote should be boosted, and the rpc would estimate the required fee based on the current boost pool liquidity

Reply:
My feeling is that we don't gain much by adding this (we can take the discussion to Linear, i've already written plenty there). We already have a boost pool depth rpc that can be used to estimate this. What do we gain by adding yet another way to get this information? 

Btw. What if boost == true and there isn't enough liquidity? Return an error?

Reply:
what we would gain is that we dont need to do the `cf_boost_pools_depth` before we can request a quote. but its not a strong opinion, we already have that implemented anyway, so its more about what might be useful for somebody else

> Btw. What if boost == true and there isn't enough liquidity? Return an error?

yes, i think would have expected a similar error to what happens if there is not enough liquidity to swap the given amount

Reply:
Do we really need to know the boost pools depth before getting a quote? At most this will change the total input amount by 0.1%, which is unlikely to impact the swap price in a any noticeable way.

To me it seems much more flexible to provide an rpc for price, and and rpc for available boost liquidity, and let people do with it what they want. We can't anticipate all the ways in which people might want to use this. For eaxmple what if we want to display both the price in boosted an unboosted scenarios?